src/payment_channel_management_strategies/PaymentChannelManagementStrategy.js

```diff
@@ -0,0 +1,59 @@
+import { find } from 'lodash';
+
+export default class PaymentChannelManagementStrategy {
+  constructor(sdkContext) {
+    this._sdkContext = sdkContext;
+  }
+
+  async selectChannel(serviceClient) {
+    const account = this._sdkContext.account;
+    await serviceClient.loadOpenChannels();
+    await serviceClient.updateChannelStates();
+    const paymentChannels = serviceClient.paymentChannels;
+    const serviceCallPrice = serviceClient.metadata.pricing.price_in_cogs;
+    const mpeBalance = await account.escrowBalance();
+    const defaultExpiration = await this._defaultChannelExpiration(serviceClient);
+
+    if(paymentChannels.length === 0) {
+      if(mpeBalance >= serviceCallPrice) {
+        return serviceClient.openChannel(serviceCallPrice, defaultExpiration);
+      }
+
+      return serviceClient.depositAndOpenChannel(serviceCallPrice, defaultExpiration);
+    }
+
+    const firstFundedValidChannel = find(paymentChannels, (paymentChanel) => this._hasSufficientFunds(paymentChanel, serviceCallPrice) && this._isValid(paymentChanel, defaultExpiration));
+    if(firstFundedValidChannel) {
+      return firstFundedValidChannel;
+    }
+
+    const firstFundedChannel = find(paymentChannels, (paymentChanel) => this._hasSufficientFunds(paymentChanel, serviceCallPrice));
+    if(firstFundedChannel) {
+      await firstFundedChannel.extendExpiration(defaultExpiration);
```

Comment:
This means the channel will be valid for calls for only 1 block, right?

Reply:
Yes, we did have a `BLOCK_OFFSET` but we decided to move that to implementation of `PaymentChannelManagementStrategy`.

[Discussion](https://github.com/singnet/snet-sdk-js/pull/1#discussion_r283896958)